tests/src/core/testqgslayoutgeopdfexport.cpp

```diff
@@ -191,27 +205,151 @@ void TestQgsLayoutGeoPdfExport::testCollectingFeatures()
+  bool expected = false;
+#else
+  bool expected = true;
+#endif
+  QCOMPARE( geoPdfExporter.finalize( QList<QgsAbstractGeoPdfExporter::ComponentLayerDetail>(), outputFile, details ), expected );
   QVERIFY( geoPdfExporter.errorMessage().isEmpty() );
 
+  QgsAbstractGeoPdfExporter::VectorComponentDetail vectorDetail;
+  for ( const auto &it :  geoPdfExporter.mVectorComponents )
+  {
+    if ( it.mapLayerId == linesLayer->id() )
+      vectorDetail = it;
+  }
+
   // read in as vector
-  std::unique_ptr< QgsVectorLayer > layer1 = qgis::make_unique< QgsVectorLayer >( QStringLiteral( "%1|layername=lines" ).arg( outputFile ),
+  std::unique_ptr< QgsVectorLayer > layer1 = qgis::make_unique< QgsVectorLayer >( QStringLiteral( "%1|layername=%2" ).arg( vectorDetail.sourceVectorPath, vectorDetail.sourceVectorLayer ),
       QStringLiteral( "lines" ), QStringLiteral( "ogr" ) );
   QVERIFY( layer1->isValid() );
   QCOMPARE( layer1->featureCount(), 6L );
-  std::unique_ptr< QgsVectorLayer > layer2 = qgis::make_unique< QgsVectorLayer >( QStringLiteral( "%1|layername=points" ).arg( outputFile ),
+  for ( const auto &it :  geoPdfExporter.mVectorComponents )
+  {
+    if ( it.mapLayerId == pointsLayer->id() )
+      vectorDetail = it;
+  }
+  std::unique_ptr< QgsVectorLayer > layer2 = qgis::make_unique< QgsVectorLayer >( QStringLiteral( "%1|layername=%2" ).arg( vectorDetail.sourceVectorPath, vectorDetail.sourceVectorLayer ),
       QStringLiteral( "lines" ), QStringLiteral( "ogr" ) );
   QVERIFY( layer2->isValid() );
   QCOMPARE( layer2->featureCount(), 32L );
-  std::unique_ptr< QgsVectorLayer > layer3 = qgis::make_unique< QgsVectorLayer >( QStringLiteral( "%1|layername=polys" ).arg( outputFile ),
+  for ( const auto &it :  geoPdfExporter.mVectorComponents )
+  {
+    if ( it.mapLayerId == polygonLayer->id() )
+      vectorDetail = it;
+  }
+  std::unique_ptr< QgsVectorLayer > layer3 = qgis::make_unique< QgsVectorLayer >( QStringLiteral( "%1|layername=%2" ).arg( vectorDetail.sourceVectorPath, vectorDetail.sourceVectorLayer ),
       QStringLiteral( "lines" ), QStringLiteral( "ogr" ) );
   QVERIFY( layer3->isValid() );
   QCOMPARE( layer3->featureCount(), 10L );
+
+
+
+  // test for theme based export here!
+  l.renderContext().setExportThemes( QStringList() << QStringLiteral( "test preset2" ) << QStringLiteral( "test preset" ) << QStringLiteral( "test preset3" ) );
+  map2->setFollowVisibilityPreset( true );
+  map2->setFollowVisibilityPresetName( QStringLiteral( "test preset3" ) );
+
+  QgsLayoutGeoPdfExporter geoPdfExporter2( &l );
+  const QString outputFile2 = geoPdfExporter2.generateTemporaryFilepath( QStringLiteral( "test_src2.pdf" ) );
+  settings = QgsLayoutExporter::PdfExportSettings();
+  settings.writeGeoPdf = true;
+  settings.exportMetadata = false;
+  exporter.exportToPdf( outputFile, settings );
+
+  // check that features were collected
+  lineFeatures = geoPdfExporter2.mCollatedFeatures.value( QString() ).value( linesLayer->id() );
```

Comment:
When exporting multiple themes, an empty group string value represents what?

Reply:
No group is associated/needed (e.g. not exporting themes)